src/arbitrary_impl.rs

```diff
@@ -0,0 +1,62 @@
+use ::arbitrary::{Arbitrary, Unstructured, Result, unstructured};
+
+use super::*;
+
+impl<'a, T, const BITS: usize> Arbitrary<'a> for UInt<T, BITS>
+where
+  T: unstructured::Int,
+  Self: Number<UnderlyingType = T>,
+{
+  fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+    u.int_in_range(Self::MIN.value()..=Self::MAX.value())
+      .map(Self::new)
+  }
+}
+
+impl<'a, T, const BITS: usize> Arbitrary<'a> for Int<T, BITS>
+where
+  T: unstructured::Int,
+  Self: SignedNumber<UnderlyingType = T>,
+{
+  fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+    u.int_in_range(Self::MIN.value()..=Self::MAX.value())
+      .map(Self::new)
+  }
+}
+
+#[allow(dead_code)]
+const fn static_assertion<'a, T>()
+where
+  T: Arbitrary<'a>,
+{}
+
+macro_rules! check {
+  ($($lit:ident), +$(,)?) => {
+    $(
+      const _: () = static_assertion::<'_, $lit>();
+    )*
+  };
+}
+
+macro_rules! check_bunch {
+  ($($start:literal..=$end:literal), +$(,)?) => {
+    $(
+      seq_macro::seq!(N in $start..=$end {
+        check!(#(u~N,)*);
+      });
+      
+      seq_macro::seq!(N in $start..=$end {
+        check!(#(i~N,)*);
+      });
+    )*
+  };
+}
+
+check_bunch!(
```

Comment:
I don't know if this applies here or not, but fyi:

u3 is an alias for `UInt<u8, 3>`. u17 is an alias for `UInt<u32, 17>`. These aliases are the typical way to use arbitrary ints, but they are not the only allowed types. For example, `UInt<u32, 3>` and even `UInt<u32, 32>` are also legal.

I don't know what this means for this implementation here, but I thought it's worth bringing this up

Reply:
Also some comments here (and in other places) would be useful. I don't really understand what this all does

Reply:
> I don't know if this applies here or not, but fyi:
> 
> u3 is an alias for `UInt<u8, 3>`. u17 is an alias for `UInt<u32, 17>`. These aliases are the typical way to use arbitrary ints, but they are not the only allowed types. For example, `UInt<u32, 3>` and even `UInt<u32, 32>` are also legal.
> 
> I don't know what this means for this implementation here, but I thought it's worth bringing this up

Ah, I forget `UInt<u8, 3>` can also be written as `UInt<u32, 3>`, I will change the implementation here.

Reply:
> Also some comments here (and in other places) would be useful. I don't really understand what this all does

The macro here is used to statically assert `Arbitrary` trait is implemented

Reply:
Why do we need to verify that the trait is implemented though? This is stated for all arbitrary-ints, right above, so what exactly does this static test add?

Something that I think would be useful to have though would be to add some actual quickcheck test. Meaning, something that uses quickcheck to test something which is the run in the CI of this crate. That would also be a usage example of why this patch is useful.

Reply:
`quickcheck` and `arbitrary` are used to generate random values, they are mostly used when implementing ser/deser framework to verify if the ser/deser implementation correct, so I am not quite sure in this crate, which functionalities want to test with the given random values by `quickcheck` and `arbitrary`.